From: Yinghai Lu <yinghai@kernel.org>
Subject: [PATCH] x86: allow NVS can be accessed by driver
References: none
Patch-Mainline: submitted, aiming for 3.2

APEI related...

Signed-off-by: Yinghai Lu <yinghai@kernel.org>
Signed-off-by: Thomas Renninger <trenn@suse.de>

Automatically created from "patches.drivers/apei_allow_drivers_access_nvs_ram.patch" by xen-port-patches.py

--- head.orig/arch/x86/kernel/e820-xen.c	2012-02-09 15:46:36.000000000 +0100
+++ head/arch/x86/kernel/e820-xen.c	2012-02-10 11:29:49.000000000 +0100
@@ -1051,7 +1051,8 @@ void __init e820_reserve_resources(void)
 		 * pcibios_resource_survey()
 		 */
 		if (e820.map[i].type != E820_RESERVED || res->start < (1ULL<<20)) {
-			res->flags |= IORESOURCE_BUSY;
+			if (e820.map[i].type != E820_NVS)
+				res->flags |= IORESOURCE_BUSY;
 			insert_resource(&iomem_resource, res);
 		}
 		res++;
